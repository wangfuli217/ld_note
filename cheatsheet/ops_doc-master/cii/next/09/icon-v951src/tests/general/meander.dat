abc:2
1234:2
ABC:4
