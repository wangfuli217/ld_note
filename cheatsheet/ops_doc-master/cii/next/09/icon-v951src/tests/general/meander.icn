#
#          M E A N D E R I N G   S T R I N G S
#

#  This main procedure accepts specifications for meandering strings
#  from standard input with the alphabet separated from the length by
#  a colon.

procedure main()
   local line, alpha, n
   while line := read() do $(
      line ? if $(
         alpha := tab(upto(':')) &
         move(1) &
         n := integer(tab(0))
         $)
         then write(meander(alpha,n)) 
         else stop("*** erroneous input ***")
      $)
end

procedure meander(alpha,n)
   local result, t, i, c, k
   i := k := *alpha
   t := n-1
   result := repl(alpha$<1$>,t)
   while c := alpha$<i$> do $(
      if find(result$<-t:0$> || c,result)
      then i -:= 1
      else $(result ||:= c; i := k$)
      $)
   return result
end
