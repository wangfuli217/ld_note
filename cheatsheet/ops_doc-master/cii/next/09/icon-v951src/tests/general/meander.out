accbcabbaa
14434241332312211
AAACCCCBCCCACCBBCCBACCABCCAACBCBCACBBBCBBACBABCBAACACABBCABACAABCAAABBBBABBAABABAAAA
